backend/carts/models.py

```diff
@@ -0,0 +1,28 @@
+from django.db import models
+from accounts.models import User
+from products.models import Product
+
+
+class Cart(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+
+class CartItem(models.Model):
+    cart = models.ForeignKey(
+        Cart, on_delete=models.CASCADE, related_name="cart_items"
+    )
+    product = models.ForeignKey(Product, on_delete=models.CASCADE)
+    quantity = models.PositiveIntegerField(default=1)
```

Comment:
Add a minimum quantity validator at the model level to avoid invalid saves outside  the serializer
```suggestion
    from django.core.validators import MinValueValidator
    quantity = models.PositiveIntegerField(default=1, validators=[MinValueValidator(1)])
```

Reply:
Thank you for the suggestion.